clients/src/main/java/org/apache/kafka/common/serialization/Serdes.java

```
@@ -0,0 +1,155 @@
+    }
+
+    static public final class DoubleSerde implements Serde<Double> {
+        @Override
+        public Serializer<Double> serializer() {
+            return new DoubleSerializer();
+        }
+
+        @Override
+        public Deserializer<Double> deserializer() {
+            return new DoubleDeserializer();
+        }
+    }
+
+    static public final class StringSerde implements Serde<String> {
+        @Override
+        public Serializer<String> serializer() {
+            return new StringSerializer();
+        }
+
+        @Override
+        public Deserializer<String> deserializer() {
+            return new StringDeserializer();
+        }
+    }
+
+    static public final class ByteArraySerde implements Serde<byte[]> {
+        @Override
+        public Serializer<byte[]> serializer() {
+            return new ByteArraySerializer();
+        }
+
+        @Override
+        public Deserializer<byte[]> deserializer() {
+            return new ByteArrayDeserializer();
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    static public <T> Serde<T> serdeFrom(Class<T> type) {
+        if (String.class.isAssignableFrom(type)) {
+            return (Serde<T>) String();
+        }
+
+        if (Integer.class.isAssignableFrom(type)) {
+            return (Serde<T>) Integer();
+        }
+
+        if (Long.class.isAssignableFrom(type)) {
+            return (Serde<T>) Long();
+        }
+
+        if (Double.class.isAssignableFrom(type)) {
+            return (Serde<T>) Double();
+        }
+
+        if (byte[].class.isAssignableFrom(type)) {
+            return (Serde<T>) ByteArray();
+        }
+
+        // TODO: we can also serializes objects of type T using generic Java serialization by default
+        throw new IllegalArgumentException("Unknown class for built-in serializer");
+    }
+
+    static public <T> Serde<T> serdeFrom(final Serializer<T> serializer, final Deserializer<T> deserializer) {
+        return new Serde<T>() {
```

There is currently the special case where either `serializer` and/or `deserializer` may be `null`.

I think in this case we should perhaps improve this code by not simply triggering, eventually, a NPE when users attempt to access the (nulled) serializer or deserializer, but rather throw an `UnsupportedOperationException`.

For example:

```java
            public Serializer<T> serializer() {
                if (serializer != null) {
                    return serializer;
                }
                else {
                     throw new UnsupportedOperationException("This serde does not support serialization");
                }

            }
```